src/temp_compare_matches_owl_and_nomenclature.py

```diff
@@ -0,0 +1,30 @@
+"""Temporary analysis file to compare exact matches to ICD11MMS between Orphanet OWL release and nomenclature files."""
+import pandas as pd
+
+from mappings import INPUT_MMS_MATCHES_TSV, xml_as_df
+
+
+# Read and clean dataframes
+nom_df = xml_as_df()
+nom_df['predicate_id'] = nom_df['DisorderMappingRelation'].apply(
+    lambda x: 'skos:exactMatch' if x.startswith('E (')
+    else 'skos:narrowMatch' if x.startswith('NTBT (')
+    else 'skos:broadMatch' if x.startswith('BTNT (')
+    else 'skos:relatedMatch')  # 0 instances of relatedMatch; just a fallback
+nom_df = nom_df[nom_df['predicate_id'] == 'skos:exactMatch'][['OrphaCode', 'Reference']]\
+    .sort_values(['OrphaCode', 'Reference']).reset_index(drop=True)
+
+owl_df = pd.read_csv(INPUT_MMS_MATCHES_TSV, sep="\t").rename(columns={
+    '?cls': 'OrphaCode',
+    '?xref': 'Reference',
+}).sort_values(['OrphaCode', 'Reference']).reset_index(drop=True)
+owl_df['OrphaCode'] = owl_df['OrphaCode'].apply(lambda x: x.split('_')[1][:-1])  # get code and remove > char
+owl_df['Reference'] = owl_df['Reference'].apply(lambda x: x.split(':')[1])  # get unprefixed code
+
+# Convert to sets of tuple edges
+set_tups = lambda df: set(zip(df['OrphaCode'], df['Reference']))
+nom_set = set_tups(nom_df)  # n=1,340
+owl_set = set_tups(owl_df)  # n=1,459
+i = nom_set.intersection(owl_set)  # n=1,323
+n_diff = nom_set.difference(owl_set)  # n=17
+o_diff = owl_set.difference(nom_set)  # n=136
```

Comment:
## Difference in exact matches between ORDO OWL and nomenclature file
I ended up not needing to query the Orphanet OWL release to get exact matches and then do lookups in the nomenclature XML. I was able to query the XML directly. I did an analysis though to compare the exact matches between the two different sources, and they are as follows.

Note that this comparison is for ICD11MMS exact matches, as the foundation references do not exist (yet) in the OWL release, but ICD11MMS refs exist in both. I was originally going to use the ICD11MMS refs as an intermediary proxy to get Orphanet->foundation refs.

I'm not too worried about the discrepancy between these sources, as the OWL is released on a 6 months schedule and the nomenclature files are on 12 months, so that is what I'm attributing the difference to.

- n exact matches in nomenclature file(s): 1,340
- n exact matches in OWL release: 1,459
- n in intersection: 1,323
- n nomenclature files(s) not in OWL: 17
- n in OWL not in nomenclature file(s): 136

Reply:
Glad this was helpful. You mis-wrote something though or possibly misunderstood something. It sounds like you're asking me to change things so that I use the OWL instead of the XML for these matches. I do agree that if that could be done, that'd be easier maintenance. But it can't be done. The OWL does not have any ICD11Foundation references. Its only potential role here would be to help us do some additional filtering on what's returned back from the XML.

In any case, given that the XML must be utilized, and given that you agree that this is a small / insignificant difference, it looks like the determination is to opt out of that additional filtering, and keep what I've done as is. So that's what I'm going with!